### DISCUSSION TITLE
usePages() is list of promises. Best way to store and read site page data?

### DISCUSSION THREAD

**Discussion by Mootook** [vuepress/core#159] [Q&A] [answered]
Hello, I recently started moving my personal blog to use vuepress-next. It's awesome!
I do have a question, however, about a use case where I need to dynamically generate some data from components (blog list, for example).

Currently, I am using the `onPrepared` hook to read each file and store some relevant data from its frontmatter in a temporarily created json file that my BlogList component reads from.

I have noticed the `usePagesData` https://github.com/vuepress/vuepress-next/blob/main/packages/%40vuepress/client/src/injections/pagesData.ts
(though it is not in the docs), and it seems to solve my use case.
It is returning an object of promises seeingly indexed by the page's id.
Is this intentional?

So for my use-case the solution is:
```ts
onBeforeMount(async () => {
  const pages = usePagesData()
  const res = await Promise.all(Object.values(pages.value).map(p => p()))
})
```

Wondering if this was just didn't get documented or is planned for removal or changes. I might switch over to using it (provided my logic correct).

Thanks again for this working on this, vuepress is really awesome!

**Comment by yzx9** [comment, marked as answer]
FYI: #72

The hook `usePagesData` is good to use  but not good for scalebility, it will course us to have to fetch all data of pages, while we only need a part of info in most cases.

**Reply by Mootook** [reply]
@nsznsznjsz  Looks like I'm on the right track then dynamically generating some json for my client code to read.
Thanks for the response!
